great_expectations/rule_based_profiler/domain_builder/cardinality_checker.py

```diff
@@ -0,0 +1,211 @@
+            return metric_value <= self._limit_mode.max_unique_values
+        elif isinstance(self._limit_mode, RelativeCardinalityLimit):
+            return metric_value <= self._limit_mode.max_proportion_unique
+
+    @staticmethod
+    def _validate_metric_value(metric_value: Union[int, float]) -> None:
+        if not isinstance(metric_value, (int, float)):
+            raise ProfilerConfigurationError(
+                f"Value of measured cardinality must be of type int or float, you provided {type(metric_value)}"
+            )
+        if metric_value < 0.00:
+            raise ProfilerConfigurationError(
+                f"Value of cardinality (number of rows or percent unique) should be greater than 0.00, your value is {metric_value}"
+            )
+
+    def _convert_to_cardinality_mode(
+        self,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[Union[int, float]] = None,
+    ) -> Union[AbsoluteCardinalityLimit, RelativeCardinalityLimit]:
+
+        self._validate_input_parameters(
+            limit_mode=limit_mode,
+            max_unique_values=max_unique_values,
+            max_proportion_unique=max_proportion_unique,
+        )
+
+        if limit_mode is not None:
+            if isinstance(limit_mode, str):
+                try:
+                    return CardinalityLimitMode[limit_mode.upper()].value
+                except KeyError:
+                    raise ProfilerConfigurationError(
+                        f"Please specify a supported cardinality mode. Supported cardinality modes are {[member.name for member in CardinalityLimitMode]}"
+                    )
+            else:
+                return limit_mode.value
+        if max_unique_values is not None:
+            return AbsoluteCardinalityLimit(
+                name=f"CUSTOM_ABS_{max_unique_values}",
+                max_unique_values=max_unique_values,
+            )
+        if max_proportion_unique is not None:
+            return RelativeCardinalityLimit(
+                name=f"CUSTOM_REL_{max_proportion_unique}",
+                max_proportion_unique=max_proportion_unique,
+            )
+
+    def _validate_input_parameters(
+        self,
+        limit_mode: Optional[Union[CardinalityLimitMode, str]] = None,
+        max_unique_values: Optional[int] = None,
+        max_proportion_unique: Optional[int] = None,
+    ) -> None:
+        params = (
+            limit_mode,
+            max_unique_values,
+            max_proportion_unique,
+        )
+        num_params: int = sum([0 if param is None else 1 for param in params])
+        if num_params != 1:
+            raise ProfilerConfigurationError(
+                f"Please pass ONE of the following parameters: limit_mode, max_unique_values, max_proportion_unique, you passed {num_params} parameters."
+            )
+
+        if limit_mode is not None:
```

Comment:
@anthonyburdi Can this be accomplished with one `if` statement using and/or logic therein (and thus without the `pass`)?

Reply:
Of course 🤦  good catch. Changed in a local commit, will push after resolving other comments.